src/backend/distributed/commands/database.c

```diff
@@ -180,15 +180,32 @@ PreprocessAlterDatabaseStmt(Node *node, const char *queryString,
 
 	AlterDatabaseStmt *stmt = castNode(AlterDatabaseStmt, node);
 
-	EnsureCoordinator();
-
 	char *sql = DeparseTreeNode((Node *) stmt);
 
-	List *commands = list_make3(DISABLE_DDL_PROPAGATION,
-								(void *) sql,
-								ENABLE_DDL_PROPAGATION);
+	if (strstr(sql, "SET TABLESPACE") != NULL)
```

Comment:
Can you use `stmt` to check this? It is not likely but what if database name was `"SET TABLESPACE"` 😄

Reply:
Fixed